src/modules/location.rs

```diff
@@ -86,30 +110,36 @@ impl Location {
 		}
 	}
 
-	async fn search_osm(client: &Client, address: &str, lang: &str) -> Result<Location> {
-		let url = format!(
-			"https://nominatim.openstreetmap.org/search?q={address}&accept-language={lang}&limit=1&format=jsonv2",
-		);
-		let results: Vec<OpenStreetMapGeoObj> = client.get(&url).send().await?.json().await?;
-		let result = results.first().ok_or_else(|| anyhow!("Location request failed."))?;
-
-		Ok(Location {
-			name: result.display_name.clone(),
-			lon: result.lon.parse::<f64>().unwrap(),
-			lat: result.lat.parse::<f64>().unwrap(),
-		})
+	async fn search_osm(client: &Client, address: &str, language: &str) -> Result<Location> {
+		client
+			.get(&OpenStreetMapLocation { address, language }.assemble())
+			.send()
+			.await?
+			.json::<Vec<OpenStreetMapGeoObj>>()
+			.await?
+			.first()
+			.ok_or_else(|| anyhow!("Location request failed."))
+			.map(|l| Location {
```

Comment:
Both APIs return a proprietary format which can be made into a `Location`.  Maybe I should move these conversions into a trait such that in the long term, there is only one querying function required -- with Rust choosing the correct behaviour based on the types.

Reply:
:wrench: :heavy_check_mark: